bpf/lib/nodeport.h

```diff
@@ -815,73 +812,69 @@ static __always_inline bool nodeport_nat_ipv4_needed(struct __ctx_buff *ctx,
 	 * overlapping tuples, e.g. applications in hostns reusing
 	 * source IPs we SNAT in node-port.
 	 */
-	if (dir == NAT_DIR_EGRESS) {
-		if (ip4->saddr == addr)
-			return true;
+	if (ip4->saddr == addr)
+		return true;
 
 #ifdef ENABLE_MASQUERADE /* SNAT local pod to world packets */
 # ifdef IS_BPF_OVERLAY
-		/* Do not MASQ when this function is executed from bpf_overlay
-		 * (IS_BPF_OVERLAY denotes this fact). Otherwise, a packet will
-		 * be SNAT'd to cilium_host IP addr.
-		 */
-		return false;
+	/* Do not MASQ when this function is executed from bpf_overlay
+	 * (IS_BPF_OVERLAY denotes this fact). Otherwise, a packet will
+	 * be SNAT'd to cilium_host IP addr.
+	 */
+	return false;
 # endif
 #ifdef IPV4_SNAT_EXCLUSION_DST_CIDR
-		/* Do not MASQ if a dst IP belongs to a pods CIDR
-		 * (native-routing-cidr if specified, otherwise local pod CIDR).
-		 * The check is performed before we determine that a packet is
-		 * sent from a local pod, as this check is cheaper than
-		 * the map lookup done in the latter check.
-		 */
-		if (ipv4_is_in_subnet(ip4->daddr, IPV4_SNAT_EXCLUSION_DST_CIDR,
-				      IPV4_SNAT_EXCLUSION_DST_CIDR_LEN))
-			return false;
+	/* Do not MASQ if a dst IP belongs to a pods CIDR
+	 * (native-routing-cidr if specified, otherwise local pod CIDR).
+	 * The check is performed before we determine that a packet is
+	 * sent from a local pod, as this check is cheaper than
+	 * the map lookup done in the latter check.
+	 */
+	if (ipv4_is_in_subnet(ip4->daddr, IPV4_SNAT_EXCLUSION_DST_CIDR,
+			      IPV4_SNAT_EXCLUSION_DST_CIDR_LEN))
+		return false;
 #endif
 
-		/* Check whether packet is from a local endpoint */
-		ep = __lookup_ip4_endpoint(ip4->saddr);
-		if (ep && !(ep->flags & ENDPOINT_F_HOST)) {
-			struct remote_endpoint_info *info;
-			*from_endpoint = true;
+	/* Check whether packet is from a local endpoint */
+	ep = __lookup_ip4_endpoint(ip4->saddr);
+	if (ep && !(ep->flags & ENDPOINT_F_HOST)) {
+		struct remote_endpoint_info *info;
+		*from_endpoint = true;
 
-			info = ipcache_lookup4(&IPCACHE_MAP, ip4->daddr,
-					       V4_CACHE_KEY_LEN);
-			if (info != NULL) {
+		info = ipcache_lookup4(&IPCACHE_MAP, ip4->daddr,
+				       V4_CACHE_KEY_LEN);
+		if (info) {
 #ifdef ENABLE_IP_MASQ_AGENT
-				/* Do not SNAT if dst belongs to any
-				 * ip-masq-agent subnet.
-				 */
-				struct lpm_v4_key pfx;
-
-				pfx.lpm.prefixlen = 32;
-				memcpy(pfx.lpm.data, &ip4->daddr, sizeof(pfx.addr));
-				if (map_lookup_elem(&IP_MASQ_AGENT_IPV4, &pfx))
-					return false;
+			/* Do not SNAT if dst belongs to any
+			 * ip-masq-agent subnet.
+			 */
```

Comment:
nit: this one should be reformatted wrt line-break since we remove an extra indent